src/Language/PureScript/Names.hs

```diff
@@ -292,14 +292,17 @@ instance ToJSON a => ToJSON (Qualified a) where
     BySourcePos ss -> toJSON2 (ss, a)
 
 instance FromJSON a => FromJSON (Qualified a) where
-  parseJSON v = byModule <|> bySourcePos
+  parseJSON v = byModule <|> bySourcePos <|> byMaybeModuleName'
     where
     byModule = do
       (mn, a) <- parseJSON2 v
       pure $ Qualified (ByModuleName mn) a
     bySourcePos = do
       (ss, a) <- parseJSON2 v
       pure $ Qualified (BySourcePos ss) a
+    byMaybeModuleName' = do
+      (mn, a) <- parseJSON2 v
+      pure $ Qualified (byMaybeModuleName mn) a
```

Comment:
Maybe this
```haskell
instance FromJSON a => FromJSON (Qualified a) where
  parseJSON v = byModule <|> bySourcePos <|> byNullSourcePos
    where
    byModule = do
      (mn, a) <- parseJSON2 v
      pure $ Qualified (ByModuleName mn) a
    bySourcePos = do
      (ss, a) <- parseJSON2 v
      pure $ Qualified (BySourcePos ss) a
    byNullSourcePos = do
      (Nothing :: Maybe SourcePos, a) <- parseJSON2 v
      pure $ Qualified ByNullSourcePos a
```
would be preferable? The `byMaybeModuleName` is redundant I think, should only reach here if the array has `null` in first position.

Reply:
Also wondering if the `ToJSON` instance should map `ByNullSourcePos` to `null` (currently maps to `[0,0]`)

Reply:
I think just updating the `SourcePos` instances seems simpler [4d70482](https://github.com/purescript/purescript/pull/4403/commits/4d7048227e1164af86647688795554bc1300b86a)